src/linear_algebra/determinant.lean

```diff
@@ -409,6 +409,25 @@ have is_unit (linear_map.to_matrix (finite_dimensional.fin_basis 𝕜 M)
     by simp only [linear_map.det_to_matrix, is_unit_iff_ne_zero.2 hf],
 linear_equiv.of_is_unit_det this
 
+lemma linear_map.associated_det_of_eq_comp (e : M ≃ₗ[R] M) (f f' : M →ₗ[R] M)
+  (h : ∀ x, f x = f' (e x)) : associated f.det f'.det :=
+begin
+  suffices : associated (f'.comp e.to_linear_map).det f'.det,
```

Comment:
```suggestion
  suffices : associated (f' ∘ₗ ↑e).det f'.det,
```
I have always imagined the simp-normal form to be `↑e`, not `e.to_linear_map`.  Do you feel differently?  I didn't test the above suggestion, but `∘ₗ` often helps elaboration with that coercion.

If you like the change, there's another opportunity in the statement of the lemma below.

Reply:
I'm always a bit unsure what the intended form is supposed to be, I thought for linear stuff the coercion is unreliable enough that `to_linear_map` is preferred.

Since it seems to work here, let's go for coercions.